### DISCUSSION TITLE
Am I billed for serverless function usage called during build time?

### DISCUSSION THREAD

**Discussion by stevenqzhang** [vercel/community#4283] [Help] [answered]
### Summary

If I use 1GB-hr of function invocation at buildtime to pregenerate pages (i.e. using `getStaticPaths`), does that count towards my serverless usage?

i.e. for https://vercel.com/dashboard/<team>/usage#Serverless%20Functions-Execution

It's not clear to me based on https://vercel.com/docs/limits/usage#execution

### Steps to Reproduce

See Summary

**Comment by amyegan** [comment, marked as answer]
Hi @stevenqzhang! Serverless Functions are not used during build time. During build, `getStaticPaths` and `getStaticProps` statically pre-render pages and would not count toward Serverless Function usage at that time.

Keep in mind that there are some scenarios in which [`getStaticProps` can run in the background](https://nextjs.org/docs/pages/building-your-application/data-fetching/get-static-props#when-does-getstaticprops-run) during runtime which impacts [Serverless Function Usage](https://vercel.com/docs/limits/usage#serverless-functions) and [Bandwidth](https://vercel.com/docs/limits/usage#bandwidth). 

> When using [Incremental Static Regeneration](https://vercel.com/docs/incremental-static-regeneration) with [Next.js](https://nextjs.org/docs/pages/building-your-application/rendering/incremental-static-regeneration), both the `revalidate` option for `getStaticProps` and `fallback` for `getStaticPaths` will result in a Function invocation.

**Reply by stevenqzhang** [reply]
great thank you!
